src/stim/stabilizers/pauli_string.pybind.cc

```diff
@@ -636,6 +585,54 @@ void stim_pybind::pybind_pauli_string_methods(pybind11::module &m, pybind11::cla
         )DOC")
             .data());
 
+    c.def_static(
+        "iter_all",
+        [](size_t num_qubits, pybind11::object &min_weight, pybind11::object &max_weight) {
+            size_t _max_weight, _min_weight;
+            if (max_weight.is_none()) {
+                _max_weight = num_qubits;
+            } else {
+                _max_weight = pybind11::cast<size_t>(max_weight);
+            }
+            if (min_weight.is_none()) {
+                _min_weight = 0;
+            } else {
+                _min_weight = pybind11::cast<size_t>(min_weight);
+            }
+            return PauliStringIterator<stim::MAX_BITWORD_WIDTH>(num_qubits, _min_weight, _max_weight);
+        },
+        pybind11::arg("num_qubits"),
+        pybind11::kw_only(),
+        pybind11::arg("min_weight") = pybind11::none(),
+        pybind11::arg("max_weight") = pybind11::none(),
```

Comment:
A useful argument here would be `allowed_paulis: str`, so the user could restrict to X errors (allowed_paulis="X") or not-Y-errors (allowed_paulis="XZ").

Reply:
Good point.

Reply:
I still need to add this, and also the random selection of signs/phases.